feat: add timestamp and message version to prompt (#944)

princeaden1 opened the pull request:
### Summary

This PR implements a timestamp feature for messages in the prompt window, responding to feature request #944.

**What this does:**
- Prefixes each sent message with a timestamp and message version.
### Screenshot

<img width="530" height="116" alt="image" src="https://github.com/user-attachments/assets/62a86890-b120-42dd-ab48-8eeb4515a292" />

princeaden1 committed:
feat: add timestamp and message version to prompt (#944)

wwwillchen commented on src/components/chat/ChatMessage.tsx:
```diff
@@ -14,13 +19,44 @@ interface ChatMessageProps {
+    const messageTimestamp = new Date(message.createdAt).getTime() / 1000; // Convert to Unix timestamp to compare with version timestamps
+    if (message.role === "user") {
+      const sortedVersions = [...liveVersions].sort(
+        (a, b) => b.timestamp - a.timestamp,
```
thanks for opening this PR!

we shouldn't need to do timestamp comparison, the assistant message itself has a commitHash.

Also, it's a bit unintuitive in terms of the UX  to show the commit hash as part of the user message, the version + timestamp should be shown as part of the assistant message.

finally, i'd prefer if it's always shown and not only when hovered to make this feature more accessible.

princeaden1 committed:
updated(chat): improve version + timestamp display for assistant messages

princeaden1 commented:
@wwwillchen Thanks for the thoughtful feedback 🙏 I've pushed the update addressing your points, version + timestamp are now always shown on assistant messages.

<img width="541" height="362" alt="image" src="https://github.com/user-attachments/assets/dd167ff3-9666-4725-937c-93bbd57e9a1d" />

wwwillchen committed:
Apply suggestion from @graphite-app[bot]

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

wwwillchen commented:
@princeaden1 - thanks, looks great!
